students/thecthaeh/Lesson08/circle_class.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+
+import math
+import functools
+
+""" A circle class that can be queried for its:
+    radius
+    diameter
+    area
+    
+    You can also print the circle, add 2 circles together, compare the size of 2 circles (which is bigger
+    or are they equal), and list and sort circles.
+    """
+
+@functools.total_ordering
+class Circle:
+    radius = 0
+    
+    def __init__(self, radius):
+        self.radius = radius
+        
+    @property
+    def diameter(self):
+        _diameter = self.radius * 2
+        return _diameter
+    
+    @diameter.setter
+    def diameter(self, value):
+        self._diameter = value
+        self.radius = round((self._diameter / 2), 2)
+    
+    @property
+    def area(self):
+        _area = (math.pi * (math.pow(self.radius, 2)))
+        return round(_area, 2)
+    
+    @classmethod
+    def from_diameter(cls, diam_value):
+        cls.diameter = diam_value
+        cls.radius = round((cls.diameter / 2), 2)
+        return cls
```

Comment:
you want to use the constructor to create the instance -> `return cls(diam_value / 2.0)`